src/ol/layer/tilelayer.js

```diff
@@ -29,14 +30,16 @@ ol.layer.TileProperty = {
  */
 ol.layer.Tile = function(opt_options) {
   var options = goog.isDef(opt_options) ? opt_options : {};
-  goog.base(this,  /** @type {olx.layer.LayerOptions} */ (options));
-
-  if (!goog.isDef(this.getPreload())) {
-    this.setPreload(0);
-  }
-  if (!goog.isDef(this.getUseInterimTilesOnError())) {
-    this.setUseInterimTilesOnError(true);
-  }
+
+  var baseOptions = goog.object.clone(options);
+
+  delete baseOptions.preload;
+  delete baseOptions.useInterimTilesOnError;
+  goog.base(this,  /** @type {olx.layer.LayerOptions} */ (baseOptions));
+
+  this.setPreload(goog.isDef(options.preload) ? options.preload : 0);
+  this.setUseInterimTilesOnError(goog.isDef(options.useInterimTilesOnError) ?
+      options.useInterimTilesOnError : true);
```

Comment:
> We'd need to add delete statements anyway, even if clone is used. No?

Correct.

I thought that deleting a non primitive type from an object will delete the copy too:
```javascript
var obj = {
  arr: [1,2,3]
};
var c = obj.arr;
delete obj.arr;
```
Unlike what I was thinking: `c == [1,2,3]`